usaspending_api/common/helpers/fiscal_year_helpers.py

```diff
@@ -111,6 +111,19 @@ def generate_fiscal_date_range(min_date: datetime, max_date: datetime, frequency
         )
         current_date = current_date + relativedelta(months=interval)
 
+    # check if max_date is in new period
+    final_period = {
+        "fiscal_year": generate_fiscal_year(max_date),
+        "fiscal_quarter": generate_fiscal_quarter(max_date),
+        "fiscal_month": generate_fiscal_month(max_date),
+    }
+    if final_period["fiscal_year"] > date_range[-1]["fiscal_year"]:
+        date_range.append(final_period)
+    elif interval == 3 and final_period["fiscal_quarter"] != date_range[-1]["fiscal_quarter"]:
```

Comment:
If I understand the logic correctly, I think this can be boiled down to a simpler conditional shown below. What am I overlooking?

```python
if final_period != date_range[-1]:
    date_range.append(final_period)
```

Reply:
I tried that, but it duplicated the final period with the wrong data depending on dates chosen

Reply:
I think I understand